src/Http/Http.Extensions/gen/RequestDelegateGenerator.cs

```diff
@@ -0,0 +1,117 @@
+            .WithTrackingName("EndpointModel");
+
+        var thunks = endpoints.Select((endpoint, _) => $$"""
+            [{{StaticRouteHandlerModelEmitter.EmitSourceKey(endpoint)}}] = (
+           (del, builder) =>
+            {
+                builder.Metadata.Add(new SourceKey{{StaticRouteHandlerModelEmitter.EmitSourceKey(endpoint)}});
+            },
+            (del, builder) =>
+            {
+                var handler = ({{StaticRouteHandlerModelEmitter.EmitHandlerDelegateType(endpoint)}})del;
+                EndpointFilterDelegate? filteredInvocation = null;
+
+                if (builder.FilterFactories.Count > 0)
+                {
+                    filteredInvocation = BuildFilterDelegate(ic =>
+                    {
+                        if (ic.HttpContext.Response.StatusCode == 400)
+                        {
+                            return System.Threading.Tasks.ValueTask.FromResult<object?>(Results.Empty);
+                        }
+                        {{StaticRouteHandlerModelEmitter.EmitFilteredInvocation()}}
+                    },
+                    builder,
+                    handler.Method);
+                }
+
+                {{StaticRouteHandlerModelEmitter.EmitRequestHandler()}}
+                {{StaticRouteHandlerModelEmitter.EmitFilteredRequestHandler()}}
+
+                return filteredInvocation is null ? RequestHandler : RequestHandlerFiltered;
+            }),
+""");
+
+        var stronglyTypedEndpointDefinitions = endpoints.Select((endpoint, _) => $$"""
+{{RequestDelegateGeneratorSources.GeneratedCodeAttribute}}
+internal static Microsoft.AspNetCore.Builder.RouteHandlerBuilder {{endpoint.HttpMethod}}(
+        this Microsoft.AspNetCore.Routing.IEndpointRouteBuilder endpoints,
+        [System.Diagnostics.CodeAnalysis.StringSyntax("Route")] string pattern,
+        {{StaticRouteHandlerModelEmitter.EmitHandlerDelegateType(endpoint)}} handler,
+        [System.Runtime.CompilerServices.CallerFilePath] string filePath = "",
+        [System.Runtime.CompilerServices.CallerLineNumber]int lineNumber = 0)
+    {
+        return MapCore(endpoints, pattern, handler, GetVerb, filePath, lineNumber);
+    }
+""");
+
+        var thunksAndEndpoints = thunks.Collect().Combine(stronglyTypedEndpointDefinitions.Collect());
+
+        context.RegisterSourceOutput(thunksAndEndpoints, (context, sources) =>
+        {
+            var (thunks, endpoints) = sources;
+
+            var endpointsCode = new StringBuilder();
+            var thunksCode = new StringBuilder();
+            foreach (var endpoint in endpoints)
+            {
+                endpointsCode.AppendLine(endpoint);
+            }
+            foreach (var thunk in thunks)
+            {
+                thunksCode.AppendLine(thunk);
+            }
+
+            var code = RequestDelegateGeneratorSources.GetGeneratedRouteBuilderExtensionsSource(
+                genericThunks: string.Empty,
```

Comment:
This is always empty string, is it needed?

Reply:
Yep, eventually we'll be generating code here to support handlers that take generic property.